### PR TITLE
Mention ReadStatTables.jl in documentation

### PR DESCRIPTION
**Pull request by junyuan-chen**
I suggest mentioning [ReadStatTables.jl](https://github.com/junyuan-chen/ReadStatTables.jl) instead of [StatFiles.jl](https://github.com/queryverse/StatFiles.jl) in the documentation.

Both packages make use of [ReadStat](https://github.com/WizardMac/ReadStat). However, StatFiles.jl does not really work well and freezes for files that are moderately large (>100MB?). It is not updated for several years and lacks features for it to be convenient in actual usage scenarios. ReadStatTables.jl has more convenience features, incorporates the development in the Julia data ecosystem and reads files much faster.

### CODE REVIEW

**Comment by bkamins** [comment]
@davidanthoff - what are your plans regarding https://github.com/queryverse/StatFiles.jl maintenance?

We can either mention both in the docs or replace StatFiles.jl with ReadStatTables.jl as @junyuan-chen suggested.

**Comment by davidanthoff** [comment]
StatFiles.jl still works, and I do have plans to get back to it at some point, so it is not abandoned. ReadStatTables.jl might well have more features, I haven't really looked into that. I'm indifferent whether both packages are mentioned or just one.

I do have some very ambitious plans for Queryverse in general, and at that point StatFiles.jl will be a key part of that. But these are plans, with no timeline, so it might well be quite a while before I go back to that work, right now I'm trying to finish up Juliaup and some VS Code extension stuff :)

**Comment by bkamins** [comment]
@davidanthoff - I am well aware of your superb work on juliaup and vscode from which the whole community benefits. I will update this PR to reflect the fact that your package is not abandoned.

I have also edited some other relevant entries.

@junyuan-chen - are you OK with the changes?

**Comment by junyuan-chen** [comment]
@bkamins These changes look good. Thank you!

File: docs/src/index.md [hunk 1]
```diff
@@ -132,11 +132,15 @@ integrated they are with DataFrames.jl.
       [GraphDataFrameBridge.jl](https://github.com/JuliaGraphs/GraphDataFrameBridge.jl)
       package.
 - **IO**:
-    - DataFrames.jl work well with a range of formats, including
-      CSVs (using [CSV.jl](https://github.com/JuliaData/CSV.jl)),
-      Apache Arrow (using [Arrow.jl](https://github.com/JuliaData/Arrow.jl))
-      Stata, SPSS, and SAS files (using [StatFiles.jl](https://github.com/queryverse/StatFiles.jl)),
-      and reading and writing parquet files (using [Parquet.jl](https://github.com/JuliaIO/Parquet.jl)).
+    - DataFrames.jl work well with a range of formats, including:
+        - CSVs (using [CSV.jl](https://github.com/JuliaData/CSV.jl)),
```
Comment (nalimilan) [inline docs/src/index.md:136]:
```suggestion
        - CSV files (using [CSV.jl](https://github.com/JuliaData/CSV.jl)),
```

File: docs/src/index.md [hunk 2]
```diff
@@ -132,11 +132,15 @@ integrated they are with DataFrames.jl.
...
+        - Apache Arrow (using [Arrow.jl](https://github.com/JuliaData/Arrow.jl))
+        - Stata, SAS and SPSS files (using [ReadStatTables.jl](https://github.com/junyuan-chen/ReadStatTables.jl);
+          alternatively [Queryverse](https://www.queryverse.org/) users
+          can choose [StatFiles.jl](https://github.com/queryverse/StatFiles.jl)),
+        - Parquet files
+          (using [Parquet2.jl](https://gitlab.com/ExpandingMan/Parquet2.jl)).
```
Comment (nalimilan) [inline docs/src/index.md:142]:
```suggestion
          (using [Parquet2.jl](https://gitlab.com/ExpandingMan/Parquet2.jl)),
```

File: docs/src/index.md [hunk 3]
```diff
@@ -132,11 +132,15 @@ integrated they are with DataFrames.jl.
...
 - **IO**:
-    - DataFrames.jl work well with a range of formats, including
-      CSVs (using [CSV.jl](https://github.com/JuliaData/CSV.jl)),
-      Apache Arrow (using [Arrow.jl](https://github.com/JuliaData/Arrow.jl))
-      Stata, SPSS, and SAS files (using [StatFiles.jl](https://github.com/queryverse/StatFiles.jl)),
-      and reading and writing parquet files (using [Parquet.jl](https://github.com/JuliaIO/Parquet.jl)).
+    - DataFrames.jl work well with a range of formats, including:
+        - CSVs (using [CSV.jl](https://github.com/JuliaData/CSV.jl)),
+        - Apache Arrow (using [Arrow.jl](https://github.com/JuliaData/Arrow.jl))
+        - Stata, SAS and SPSS files (using [ReadStatTables.jl](https://github.com/junyuan-chen/ReadStatTables.jl);
```
Comment (nalimilan) [inline docs/src/index.md:138]:
```suggestion
        - reading Stata, SAS and SPSS files (using [ReadStatTables.jl](https://github.com/junyuan-chen/ReadStatTables.jl);
```

**Review by nalimilan — APPROVED** [review]

File: docs/src/index.md [hunk 4]
```diff
@@ -132,11 +132,15 @@ integrated they are with DataFrames.jl.
...
+          alternatively [Queryverse](https://www.queryverse.org/) users
+          can choose [StatFiles.jl](https://github.com/queryverse/StatFiles.jl)),
+        - Parquet files
+          (using [Parquet2.jl](https://gitlab.com/ExpandingMan/Parquet2.jl)).
+        - reading R data files (.rda, .RData) (using [RData.jl](https://github.com/JuliaData/RData.jl))
```
Comment (bkamins) [inline docs/src/index.md:143]:
```suggestion
        - reading R data files (.rda, .RData) (using [RData.jl](https://github.com/JuliaData/RData.jl)).
```

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**